src/pages/FacilityOrganization/FacilityOrganizationUsers.tsx

```diff
@@ -1,47 +1,56 @@
 import { useQuery } from "@tanstack/react-query";
-import { useQueryParams } from "raviger";
 import { useTranslation } from "react-i18next";
 
 import CareIcon from "@/CAREUI/icons/CareIcon";
 
 import { Button } from "@/components/ui/button";
 import { Card, CardContent } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
 
 import { Avatar } from "@/components/Common/Avatar";
-import {
-  CardGridSkeleton,
-  CardListSkeleton,
-} from "@/components/Common/SkeletonLoading";
+import { CardGridSkeleton } from "@/components/Common/SkeletonLoading";
 import { UserStatusIndicator } from "@/components/Users/UserListAndCard";
 
+import useFilters from "@/hooks/useFilters";
+
 import routes from "@/Utils/request/api";
 import query from "@/Utils/request/query";
+import EditFacilityUserRoleSheet from "@/pages/FacilityOrganization/components/EditFacilityUserRoleSheet";
+import FacilityOrganizationLayout from "@/pages/FacilityOrganization/components/FacilityOrganizationLayout";
+import LinkFacilityUserSheet from "@/pages/FacilityOrganization/components/LinkFacilityUserSheet";
 import AddUserSheet from "@/pages/Organization/components/AddUserSheet";
 import { OrganizationUserRole } from "@/types/organization/organization";
 
-import EditFacilityUserRoleSheet from "./components/EditFacilityUserRoleSheet";
-import FacilityOrganizationLayout from "./components/FacilityOrganizationLayout";
-import LinkFacilityUserSheet from "./components/LinkFacilityUserSheet";
-
 interface Props {
   id: string;
   facilityId: string;
 }
 
 export default function FacilityOrganizationUsers({ id, facilityId }: Props) {
-  const [qParams, setQueryParams] = useQueryParams<{
-    sheet: string;
-    username: string;
-  }>();
+  const { qParams, updateQuery, Pagination, resultsPerPage } = useFilters({
+    limit: 15,
+    cacheBlacklist: ["search"],
+  });
   const { t } = useTranslation();
 
   const openAddUserSheet = qParams.sheet === "add";
   const openLinkUserSheet = qParams.sheet === "link";
 
   const { data: users, isLoading: isLoadingUsers } = useQuery({
-    queryKey: ["facilityOrganizationUsers", facilityId, id],
-    queryFn: query(routes.facilityOrganization.listUsers, {
+    queryKey: [
+      "facilityOrganizationUsers",
+      facilityId,
+      id,
+      qParams.search,
+      qParams.page,
+    ],
+    queryFn: query.debounced(routes.facilityOrganization.listUsers, {
       pathParams: { facilityId, organizationId: id },
+      queryParams: {
+        username: qParams.search,
+        limit: resultsPerPage,
+        offset: (qParams.page - 1) * resultsPerPage,
```

Comment:
```suggestion
        offset: ((qParams.page ?? 1) - 1) * resultsPerPage
```
qParams.page will be undefined for the first page.